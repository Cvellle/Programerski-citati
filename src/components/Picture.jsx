import React from 'react'
import './Picture.css'

const Picture =props=>{
  return(
    <div className="thumbnail">
      <h3>{props.autor}</h3>
        <img  src={props.slika} alt="Loading..."/>
      </div>
    );
}
export default Picture
