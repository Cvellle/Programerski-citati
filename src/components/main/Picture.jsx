import React from 'react'
import './Picture.css'

const Picture = ({ author, imgSrc }) => {
  if (!imgSrc) return ''
  return (
    <div className="thumbnail">
      <h3>{author}</h3>
      <img className="main-image" src={imgSrc} alt={author} />
    </div>
  )
}
export default Picture
