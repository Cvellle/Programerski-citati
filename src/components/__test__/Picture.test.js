import React from 'react';
import { shallow } from 'enzyme';
import Picture from '../Picture';

describe('Component: Picture snapshot', () => {
    it('should render Picture ', () => {
        const wrapper = shallow(<Picture author="" imgSrc="" />);

        expect(wrapper).toMatchSnapshot();
    });
});

describe('Picture component', () => {
    const wrapper = shallow(<Picture author="Autor" imgSrc="src" />);

    const author = wrapper.find("h3");
    const image = wrapper.find("img");

    test('Picture has author', () => {
        expect(author.exists()).toBeTruthy();
    });

    test('Picture has image', () => {
        expect(image.exists()).toBeTruthy();
    });
});
