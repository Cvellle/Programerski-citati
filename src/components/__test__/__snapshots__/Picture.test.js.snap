// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Picture snapshot should render Picture  1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Picture
    author=""
    imgSrc=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): "",
  Symbol(enzyme.__nodes__): Array [
    "",
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
